db.js

```diff
@@ -272,23 +269,65 @@ export class ActivityPubDB extends EventTarget {
     }
 
     const index = tx.store.index(indexName)
-    let cursor = await index.openCursor(keyRange)
 
-    const notes = []
-    while (cursor) {
-      notes.push(cursor.value)
-      cursor = await cursor.continue()
-    }
+    // For random sort
+    if (sort === 0) {
+      const totalNotes = await index.count()
+      for (let i = 0; i < limit; i++) {
```

Comment:
Make a var called `count` and change this to `while (count < limit)` and increment `count` if you have a successful yield. Else we'd return the wrong number of items